.travis.yml

```diff
@@ -10,48 +10,124 @@ env:
     - GROUP=deploy.razee.io
     - VERSION=v1alpha2
 
-script:
-  # Audit npm packages. Fail build whan a PR audit fails, otherwise report the vulnerability and proceed.
-  - if [ "${TRAVIS_PULL_REQUEST}" != "false" ]; then npm audit; else npm audit || true; fi
-  - npm run lint
-  - npm test
-  - docker build --rm -t "quay.io/razee/remoteresources3:${TRAVIS_COMMIT}" .
-  - if [ -n "${TRAVIS_TAG}" ]; then docker tag quay.io/razee/remoteresources3:${TRAVIS_COMMIT} quay.io/razee/remoteresources3:${TRAVIS_TAG}; fi
-  - if [[ $TRAVIS_TAG =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then npm version --no-git-tag-version "${TRAVIS_TAG}"; fi
-  - docker images
-  - ./build/process-template.sh kubernetes/RemoteResourceS3/resource.yaml >/tmp/resource.yaml
+jobs:
+  include:
+   - stage: building amd64 image
+     arch: amd64
+     script:
+       # Audit npm packages. Fail build whan a PR audit fails, otherwise report the vulnerability and proceed.
+       - if [ "${TRAVIS_PULL_REQUEST}" != "false" ]; then npm audit; else npm audit || true; fi
+       - npm run lint
+       - npm test
+       - docker build --rm -t "quay.io/razee/remoteresources3:${TRAVIS_COMMIT}-amd64" .
+       - if [ -n "${TRAVIS_TAG}" ]; then docker tag quay.io/razee/remoteresources3:${TRAVIS_COMMIT}-amd64 quay.io/razee/remoteresources3:${TRAVIS_TAG}-amd64; fi
+       - if [[ $TRAVIS_TAG =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then npm version --no-git-tag-version "${TRAVIS_TAG}"; fi
+       - docker images
+       - ./build/process-template.sh kubernetes/RemoteResourceS3/resource.yaml >/tmp/resource.yaml
+ 
+     before_deploy:
+       - docker login -u="${QUAY_ID}" -p="${QUAY_TOKEN}" quay.io
+       
+     deploy:
+       # Deploy alpha builds
+       - provider: script
+         script: docker push "quay.io/razee/remoteresources3:${TRAVIS_TAG}-amd64"
+         skip_cleanup: true
+         on:
+           tags: true
+           condition: ${TRAVIS_TAG} =~ ^[0-9]+\.[0-9]+\.[0-9]+_[0-9]{3}$
+       # Deploy released builds
+       - provider: script
+         script: docker push "quay.io/razee/remoteresources3:${TRAVIS_TAG}-amd64"
+         skip_cleanup: true
+         on:
+           tags: true
+           condition: ${TRAVIS_TAG} =~ ^[0-9]+\.[0-9]+\.[0-9]+$
+       - provider: releases
+         file: /tmp/resource.yaml
+         skip_cleanup: true
+         api_key: "${GITHUB_TOKEN}"
+         on:
+           tags: true
+           condition: ${TRAVIS_TAG} =~ ^[0-9]+\.[0-9]+\.[0-9]+$
+       - provider: npm
+         email: "${NPMJS_EMAIL}"
+         api_key: "${NPMJS_API_KEY}"
+         name: "${TRAVIS_TAG}"
+         skip_cleanup: true
+         on:
+           tags: true
+           condition: ${TRAVIS_TAG} =~ ^[0-9]+\.[0-9]+\.[0-9]+$      
```

Comment:
i think this section and line 93-107 (second stage) need to be put into the final stage, instead of running twice in these 2 prior stages. Otherwise i think npm will error out on the second stage since the tag will have already been created here in the first stage.

Reply:
thanks Alex, have made the required changes